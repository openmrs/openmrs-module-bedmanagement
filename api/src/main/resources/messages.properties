${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags