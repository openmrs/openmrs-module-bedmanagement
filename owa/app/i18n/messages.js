export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        ROW_REQUIRED_MSG: 'Row is required field',
        COLUMN_REQUIRED_MSG: 'Column is required field',
        ROW_SHOULD_GREATER_THAN_ZERO: 'Row value should be greater than 0',
        ROW_SHOULD_GREATER_THAN_LAYOUT_ROW: 'Row value should not be greater than layout row size',
        COLUMN_SHOULD_GREATER_THAN_ZERO: 'Column value should be greater than 0 and less than 11',
        COLUMN_SHOULD_GREATER_THAN_LAYOUT_COLUMN: 'Row value should not be greater than layout row size',
        FIX_ERROR_MSG: 'Fix error before submit',
        ADMISSION_LOCATION_SAVE_MSG: 'Admission location save successfully',
        BED_LAYOUT_SAVE_MSG: 'Admission location bed layout save successfully',
        BED_TAG_SAVE_MSG: 'Bed Tag save successfully',
        BED_TYPE_SAVE_MSG: 'Bed Type save successfully',
        BED_SAVE_MSG: 'Bed save successfully',
        DELETE_ADMISSION_LOCATION_CONFIRM_MESSAGE:
            'Are you sure you want to delete admission location {location_name}?',
        DELETE_BED_CONFIRM_MESSAGE: 'Are you sure you want to delete bed number {bed_number}?',
        DELETE_BED_TAG_CONFIRM_MSG: 'Are you sure you want to delete bed Tag {bed_tag_name}?',
        DELETE_BED_TYPE_CONFIRM_MSG: 'Are you sure you want to delete bed type {bed_type_name}?',
        DELETE_SUCCESSFULLY: 'Delete successfully',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        SAVE: 'Save',
        SAVING: 'Saving...',
        CANCEL: 'Cancel',
        ADD: 'Add',
        EDIT: 'Edit',
        DELETE: 'Delete',
        ADD_BED: 'Add Bed',
        ROWS: 'Rows',
        ROW: 'Row',
        COLUMNS: 'Columns',
        COLUMN: 'Column',
        SET_LAYOUT: 'Set Layout',
        LOCATION: 'Location',
        BED: 'Bed',
        BED_NUMBER: 'Bed Number',
        BED_TYPE: 'Bed Type',
        WARD: 'Ward',
        PARENT_LOCATION: 'Parent Location',
        NONE: 'None',
        NAME: 'Name',
        DESCRIPTION: 'Description',
        DISPLAY_NAME: 'Display Name',
        ACTION: 'Action',
        ADD_NEW: 'Add New',
        BED_TAG: 'Bed Tag',
        EXISTING_BED_TYPES: 'Existing Bed Types',
        EXISTING_BED_TAGS: 'Existing Bed Tags'
    }
};
