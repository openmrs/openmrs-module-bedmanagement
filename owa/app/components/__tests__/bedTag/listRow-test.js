import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';
import PropTypes from 'prop-types';

import BedTagListRow from 'components/bedTag/bedTagList/bedTagListRow';
import bedTagFunctionsMock from 'components/__mocks__/bedTagFunctions-mock';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        bedTagFunctions: bedTagFunctionsMock
    },
    context: {
        intl: intl
    }
};

describe('BedTagListRow', () => {
    it('Should render bed tag list properly', () => {
        const bedTagListRow = shallow(
            <BedTagListRow
                bedTag={testData.props.bedTagFunctions.getBedTagByUuid('ff7ed494-7b9c-4478-812a-5187e297f94c')}
                bedTagFunctions={testData.props.bedTagFunctions}
            />,
            {context: testData.context}
        );

        expect(
            bedTagListRow
                .find('td')
                .at(0)
                .text()
                .trim()
        ).toBe('Isolation');
        expect(shallowToJson(bedTagListRow)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnDeleteHandler = jest.spyOn(BedTagListRow.prototype, 'deleteHandler');
        const spyOnEditHandler = jest.spyOn(BedTagListRow.prototype, 'editHandler');
        const spyOnSetState = jest.spyOn(testData.props.bedTagFunctions, 'setState');
        const WrapperBedTagListRow = () => {
            return (
                <table>
                    <tbody>
                        <BedTagListRow
                            bedTag={testData.props.bedTagFunctions.getBedTagByUuid(
                                'ff7ed494-7b9c-4478-812a-5187e297f94c'
                            )}
                            bedTagFunctions={testData.props.bedTagFunctions}
                        />
                    </tbody>
                </table>
            );
        };

        WrapperBedTagListRow.contextTypes = {
            intl: PropTypes.object
        };

        const bedTagList = mount(
            <WrapperBedTagListRow
                bedTag={testData.props.bedTagFunctions.getBedTagByUuid('ff7ed494-7b9c-4478-812a-5187e297f94c')}
                bedTagFunctions={testData.props.bedTagFunctions}
            />,
            {context: testData.context}
        );

        bedTagList.find('.fa-edit').simulate('click');
        expect(spyOnEditHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'addEdit',
            pageData: {
                operation: 'edit',
                bedTagUuid: 'ff7ed494-7b9c-4478-812a-5187e297f94c'
            }
        });

        bedTagList.find('.fa-trash').simulate('click');
        expect(spyOnDeleteHandler).toHaveBeenCalled();
    });
});
