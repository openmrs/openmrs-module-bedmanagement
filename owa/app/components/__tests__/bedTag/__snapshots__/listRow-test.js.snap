// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedTagListRow Should render bed tag list properly 1`] = `
<tr>
  <td>
    Isolation
  </td>
  <td />
  <td>
    <a
      href="javascript:void(0);"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="icon fa fa-edit"
      />
       
      Edit
    </a>
      |  
    <a
      href="javascript:void(0);"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="icon fa fa-trash"
      />
       
      Delete
    </a>
  </td>
</tr>
`;
